File: JavaScript/AoC2025/days/day2.js
import fs from "fs"
import path from "path"

export default class Day2 {
    constructor() {
        const filePath = path.resolve("../../Data/2025/day2.txt");
        const data = fs.readFileSync(filePath, "utf-8");
        this.lines = data.trim().split(",");
        this.doubleRegex = /^(\d+)\1$/;
        this.multipleRegex = /^(\d+)\1+$/;
    }

    printResults() {
        console.log("Day 2:");
        console.log(`Part 1: ${this.part1()}`);
        console.log(`Part 2: ${this.part2()}`);
    }

    part1() {
        let total = 0;
        for (const line of this.lines) {
            const boundaries = line.split("-");
            const start = parseInt(boundaries[0]);
            const end = parseInt(boundaries[1]);

            for (let i = start; i <= end; i++) {
                if (this.doubleRegex.test(i.toString()))
                    total += i;
            }
        }
        return total;
    }

    part2() {
        let total = 0;
        for (const line of this.lines) {
            const boundaries = line.split("-");
            const start = parseInt(boundaries[0]);
            const end = parseInt(boundaries[1]);

            for (let i = start; i <= end; i++) {
                if (this.multipleRegex.test(i.toString()))
                    total += i;
            }
        }
        return total;
    }
}

File: Python/AoC2025/days/day2.py
import re

class Day2:
    def __init__(self):
        with open("../../Data/2025/day2.txt") as f:
            self.data = f.readlines()
        self.ranges = self.parse_input()
        self.double_regex = r"^(\d+)\1$"
        self.multiple_regex = r"^(\d+)\1+$"

    def solution(self):
        total1 = 0
        total2 = 0
        for r in self.ranges:
            boundaries = r.split("-")
            start, end = int(boundaries[0]), int(boundaries[1])
            for num in range(start, end + 1):
                if re.fullmatch(self.double_regex, str(num)):
                    total1 += num
                if re.fullmatch(self.multiple_regex, str(num)):
                    total2 += num
        print("Day 2:")
        print(f"Part 1: {total1}")
        print(f"Part 2: {total2}")

    def parse_input(self):
        ranges = self.data[0].split(",")
        return ranges
